WordPress/gutenberg
# Performance tests: Make theme versions consistent cross-env

## What?

Make performance testing consistent cross-env theme-wise.

_This is an alternative approach to https://github.com/WordPress/gutenberg/pull/50868 (See https://github.com/WordPress/gutenberg/pull/50868#issuecomment-1558989101)_


## Why?

[For consistency sake](https://github.com/WordPress/gutenberg/pull/47037/files#r1087561588), we've been using [fixed versions](https://github.com/WordPress/gutenberg/blob/trunk/bin/plugin/commands/performance.js#L340-L341) of `twentytwentyone` and `twentytwentythree` themes in our performance tests. The way they are currently being activated, though, is applied only to the `performance.js` runner (which we use in CI). The same tests, when run directly via `npm run test:performance <suite-name>`, will not use the same themes because they're not available via the [root wp-env instance](https://github.com/WordPress/gutenberg/blob/trunk/.wp-env.json) – they will use the latest versions instead.


## How?

As with other test themes, we can upload the target versions to the `gutenberg-test-themes` folder and utilize them via test env folder mapping in the wp-env config.

_For more details see @oandregal's comment: https://github.com/WordPress/gutenberg/pull/50905#discussion_r1205181738_


## Testing Instructions

- CI perf tests should pass,
- The following should pass locally:
   - `npm run test:performance front-end-block-theme.test.js`
   - `npm run test:performance front-end-classic-theme.test.js`
   - `npm run test:performance site-editor.test.js`
   - `npm run test:performance post-editor.test.js`

- The following suites should be using the following themes:
   - `front-end-block-theme.test.js` → `twentytwentythree@v1.0`
   - `front-end-classic-theme.test.js` → `twentytwentyone@v1.7`
   - `site-editor.test.js` → `emptytheme`
   - `post-editor.test.js` → `emptytheme`
